src/utils.jl

```diff
@@ -349,17 +349,21 @@ function makeunique(names)
     set = Set(names)
     length(set) == length(names) && return Symbol[Symbol(x) for x in names]
     nms = Symbol[]
+    nmsset = Set{eltype(names)}()
```

Comment:
It seems to me, without having actually tried to code it out, that we could avoid needing this separate `nmsset` and just re-use the `set` from above? And then we would check `if nm in set` below, then just `while newnm in set`, then finally do `push!(set, nm)`. That way we avoid having two `Set`s that basically do the same thing?

I remember this code being a little subtle in a few cases though, so perhaps I'm forgetting an oddity that prevents us doing it that way.

Reply:
I was also thinking about that, but that would change how the columns are named in some cases. And this could break existing code relying on the uniquified names.

Consider the following code:
```julia
CSV.makeunique([:a :a])
```
Without `nmsset`, as you propose, the result is:
```
2-element Vector{Symbol}:
 :a_1
 :a_2
```
while with `nmsset` it is:
```
2-element Vector{Symbol}:
 :a
 :a_1
```

Reply:
It seems it will be possible to use `suffixes` Dict instead of `nmsset`. The result will be slightly shorter code. I'm currently testing it and push it soon.